pkg/dbconn/kill.go

```diff
@@ -225,8 +220,9 @@ func GetLockingTransactions(ctx context.Context, db *sql.DB, tables []*table.Tab
 func GetTableLocks(ctx context.Context, db *sql.DB, tables []*table.TableInfo, logger loggers.Advanced, ignorePIDs []int) ([]*LockDetail, error) {
 	tx, err := db.BeginTx(ctx, nil)
 	if err != nil {
-		logger.Errorf("failed to begin transaction: %v", err)
+		return nil, err
```

Comment:
This is likely a bug. If it can't get a transaction it should return an error. I haven't reproduced it yet.